src/redux/saga/user.ts

```diff
@@ -0,0 +1,53 @@
+import { put, takeLatest } from 'redux-saga/effects'
+import axios from 'axios'
+
+import LocalStorage from '../../local-storage'
+import { LoginUserRequestAction } from '../types'
+
+import { loginUserSuccess, loginUserFail } from '../actions/user'
+
+// function* loginUserSaga() {
+//   try {
+//     const credentialData = yield select(credential)
+//     const res = yield axios.post('/login/local', {
+//       email: credentialData.email,
+//       password: credentialData.password,
+//     })
+//     console.log('res', res)
+
+//     yield put(loginUserSuccess(res.data.payload))
+//     yield LocalStorage.saveToken(res.data.payload.token)
+//   } catch (error) {
+//     yield put(loginUserFail())
+//   }
+// }
+
+function* loginUserSaga(action: LoginUserRequestAction) {
+  try {
+    const email = yield action.payload.credential.email
+    const password = yield action.payload.credential.password
+    const history = yield action.payload.history
```

Comment:
Move this outside the `try/catch` so you don't have to repeat it two times

Reply:
:+1: